### ISSUE TITLE
Issues with location names in (traditional) Chinese showing as simplified

### ISSUE THREAD

**Issue by null-von-sushi** [organicmaps/organicmaps#8723]
**Describe the issue**
When the language is set to Chinese traditional (Taiwan), some place names are shown in simplified script instead.

**Steps to reproduce**
1. Set the per-app language to 繁體中文（台灣）in Android (14) system settings.
2. Zoom into London, accept to download the map data if prompted.
4. Find Haidilao Hotpot (near Piccadilly Circus station).
5. Observe the erroneous use of simplified characters instead of `海底撈火鍋` for the place name.

**Expected behaviour**
As the display language is not set to simplified chinese, the location names should be displayed in normal script with local language (English in this case) as a fallback

**Screenshots**
![Screenshot_20240718-145250](https://github.com/user-attachments/assets/eba63738-9360-4a29-b1a0-2a4596b39c90)
![Screenshot_20240718-143813](https://github.com/user-attachments/assets/0b678f82-4435-4e60-90b9-5564616cb687)

**System information:**
 - Operating system and its version: Graphene OS (build 2024071600), Android 14
 - Organic Maps version: 2024.07.08-3-Google
 - Device Model: Google Pixel 6 Pro

**Additional context**
First issue here, and also kinda the first time interacting with OSM in any way. 
I found that some places show as simplified Chinese for some reason. [Haidilao for example does have an ZH_Hant entry that's correct](https://www.openstreetmap.org/node/6255364830), but for some reason it still shows simplified in-app? It's really inconsistent and hard to figure out tbh what's going on to be honest. _London_ itself is also wrong, however _City of London_ shows correctly? But then the website for OSM seems to [not have _any_ Chinese translation for neither _London_?](https://www.openstreetmap.org/relation/65606) nor for [the _City of London?_](https://www.openstreetmap.org/relation/51800).

I do understand that OSM is user submitted data, and some places only have one or the other (or for some reason, only having `zh` instead of  `zh_hant` / `zh_hans`), however Haidilao is a perfect example of all tags existing, and yet it shows incorrectly. For the two London examples, I am not sure what's going on.

As for related bugs, I found #8478 was closed as fixed, but from what I can tell the commit is only about making TTS more idiomatic? 

Apologies in advance for any inaccuracies or misunderstandings.

**Comment by null-von-sushi** [comment]
As a sidenote, if one tries to edit a location from within Organic Maps, only "Chinese" is shows as opposed to both traditional and simplified being shown, so there's no way to actually contribute useful Chinese translations from within Organic Maps it seems (I guess 'Chinese' will only submit `zh` data?)

**Comment by biodranik** [comment]
Only zh (Simplified) is stored now in OM data. There are several options:
1. Add support for zh-Hant (we have a maximum of 64 supported languages and only a few language slots are available now, supporting more requires a total refactoring of the map data format).
2. Transliterate Simplified to Traditional using ICU (with caching, of course): https://unicode-org.github.io/icu/userguide/transforms/general/#scriptlanguage

**Comment by null-von-sushi** [comment]
If transliterating, I highly suggest doing traditional to simplified instead of the other way around, as simplified has merged some character into one which may cause issues later on. It's a lot easier to map traditional to simplified than the other way around.

But just so I understand, this is an issue with Organic Maps specifically, not OpenStreetMap the service itself?

**Comment by olivertzeng** [comment]
Same issue here

**Comment by olivertzeng** [comment]
Hey @biodranik any progress on this?

![image](https://github.com/user-attachments/assets/311bd361-031f-43b0-912f-59a4de173b3b)

Furthermore some shows Simplified Chinese/Traditional Chinese

Which looks even worse

I really hope that this issue gets fixed

**Comment by biodranik** [comment]
@olivertzeng please check OpenStreetMap.org map data when you see both mixed, they are usually both added to the database as one name or name:zh* string.

**Comment by olivertzeng** [comment]
> @olivertzeng please check OpenStreetMap.org map data when you see both mixed, they are usually both added to the database as one name or name:zh* string.
> 
> 

I think maybe there's a way to distinguish between two fields if it's traditional Chinese or simplified Chinese and only show one but currently I can't think of a solution if it's not separated between zh-tw and zh-cn already

**Comment by olivertzeng** [comment]
![image](https://github.com/user-attachments/assets/42c4018a-26d5-4928-ad02-17cd87d7767a)

It looks so messy rn

With three languages and even different separators I don't know why it is like that

I would suggest if there's multiple entries either only show one like I said or show English/the original language

It clutters too much

Also I cannot figure out how to properly select a whole country to see the details of it, can you show me how the zh-* tags are managed?

**Comment by olivertzeng** [comment]
Thought to ping you @biodranik

**Comment by biodranik** [comment]
@olivertzeng isn't it an OpenStreetMap.org map data issues? I think these names are stored there in different languages in the same `name` tag.

**Comment by olivertzeng** [comment]
> @olivertzeng isn't it an OpenStreetMap.org map data issues? I think these names are stored there in different languages in the same `name` tag.

@biodranik i don't know how to check a country's name tag since I cannot really select and see the information of a country in either Go!! Maps and openstreetmap.org

Can you show me Slovenija as an example of what the name:zh-tw name:zh-cn and name:zh-hk is?

Also assuming all of these tags exist why if for example I choose Traditional Chinese(Taiwan) it shows name:zh instead of name:zh-tw?

Does it simply not exist?

**Comment by null-von-sushi** [comment]
I am not sure I understand the recent chain of comments, but AFAIK Oranic Maps unfortunately uses zh, which is messy since it's a wild west of whatever people enter first. Normally the frontend should prioritize the regional tags first, but unfortunately Organic Maps always takes zh. 

<img width="2292" height="66" alt="Image" src="https://github.com/user-attachments/assets/287240d4-c618-4237-8a7a-c55feda236d0" />

I really think that the source of the 'we can only support so many languages' should be dealt with, or just have another version with different languages (like, idk, one for the Western languages, one for Eastern languages? Maybe having two build targets with a different data URL is easier than refactoring the whole app?)

Or.....just use use ZH_Hant as the default (because traditional -> simplified is easier than the other way around) and at least it's mandarin (which is a tad more international than cantonese, sorry, though if we could I definitely think Cantonese should be supported too ❤️)

<img width="328" height="170" alt="Image" src="https://github.com/user-attachments/assets/7a6190e1-c9b6-4b8d-8e99-8ca42159c229" />

The official openstreetmap wiki even mentions that it's better to use "it is better to use name:zh-Hant=* and name:zh-Hans=* to represent the name written in traditional and simplified Chinese than the ambiguous name:zh=*."

**Comment by biodranik** [comment]
You're right, the proper support of different languages is needed. As we're now hard-limited by the format restriction to up to 64 languages we introduced in 2010, that's a tricky part, but we'll find a solution.

**Comment by olivertzeng** [comment]
Hey I just found out that according to go maps name:tw is nearly and mostly but not quite equivalent to name:zh_Hant

![image](https://github.com/user-attachments/assets/07a73b60-caa6-49a5-9af2-661b344f67b6)

**Comment by olivertzeng** [comment]
I've thought of a way to maybe implement this

If set to traditional Chinese Taiwan first search and use name:tw then name:zh_Hant and lastly name:zh that would give the best result

Same with simplified Chinese which name:zh_Hans priority > name:zh
